docs/content/sips/022-modularizing-spin-manifests.md

```diff
@@ -0,0 +1,264 @@
+
+### One component per component manifest
+
+Similar to other tools such as cargo or npm, under this proposal Spin would only support a single component to be defined in each sub-manifest. This allows reducing boilerplate in each manifest, because now sections such as `[build]` are unambiguously scoped and don’t need to be expanded to `[component.foo.build]`. It also allows grasping the application’s general structure just by looking at the application manifest, instead of having to look at each sub-manifest to find the full list of components.
+
+*Note that this doesn’t preclude future additions for defining nested components visible only in the scope of an individual top-level component, and destined for composition into said top-level component.*
+
+### Support for defining multiple components in the application manifest
+
+This proposal would not take away support for defining components using the existing syntax; it’d purely add support for the modular approach. The reason is that there are component definitions that are very light-weight; moving them into separate folders would add boilerplate and cognitive overhead, not reduce it.
+
+### Inheriting definitions from the application manifest
+
+Similar to [crates in a Cargo workspace](https://doc.rust-lang.org/1.85.1/cargo/reference/workspaces.html#the-package-table), component manifests can inherit certain definitions from the application manifest, such as the version, description, and authors:
+
+```toml
+# [project]/Spin.toml
+
+[application]
+version = "0.1.0"
+...
+
+# [project]/my-component/Spin.toml
+...
+# Inherit the `version` field from the application
+version.application = true
+```
+
+### Handling of variables
+
+Variables are among the most tricky aspects to sort out in this approach. The proposal aims to make it easy to use both application-wide and component-scoped variables. An additional design goal is to support adoption without changes to deployment pipelines and their handling of variables.
+
+To achieve these goals, the proposal
+
+- treats application-level variables as inheritable in the same way as the definitions mentioned above
+- namespaces component-level variables by prefixing them with the component name when interpreting manifests
+
+To illustrate this, consider these manifests:
+
+```toml
+# [project]/Spin.toml
+
+...
+[variables]
+my_variable = { default = "my-value" }
+...
+
+components = ["my-component"]
+```
+
+```toml
+# [project]/my-component/Spin.toml
+
+...
+# Inherit the `my_variable` variable from the application
+[variables]
+my_variable.application = true
+my_other_variable = { default = "some-value" }
+...
+```
+
+Combined, they’d be interpreted like this:
+
+```toml
+...
+[variables]
+my_variable = { default = "my-value" }
+my_component_my_other_variable = { default = "some-value" }
+...
+```
+
+Any uses in template strings are adapted to use the prefixed variable names.
```

Comment:
Is this saying that Spin has to rewrite template strings to the new names as part of lockfile creation?

How does this affect providers?  Do secrets in Vault need to use the scoped name?